provision/rosa-cross-dc/Taskfile.yml

```diff
@@ -0,0 +1,308 @@
+
+dotenv: [ '.env' ]
+
+includes:
+  common: ../common
+  ispn:
+    taskfile: ../infinispan/Utils.yaml
+    dir: ../infinispan/
+
+tasks:
+
+  create-aurora:
+    desc: "Creates a Aurora Database"
+    requires:
+      vars:
+        - AURORA_CLUSTER
+        - AURORA_REGION
+    env:
+      AURORA_CLUSTER: "{{.AURORA_CLUSTER}}"
+      AURORA_REGION: "{{.AURORA_REGION}}"
+    cmds:
+      - echo "Deploying Aurora Database ({{.AURORA_CLUSTER}}) in region {{.AURORA_REGION}}"
+      - ../aws/rds/aurora_create.sh
+
+  create-peering-connection:
+    desc: "Creates Aurora Peering Connection"
+    internal: true
+    silent: false
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME
+        - AURORA_CLUSTER
+        - AURORA_REGION
+    cmd: CLUSTER_NAME="{{.ROSA_CLUSTER_NAME}}" KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}" SKIP_ROSA_LOGIN=true {{.RDS_DIR}}/aurora_create_peering_connection.sh
+    preconditions:
+      - test -f {{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}
+
+  fetch-aurora-endpoint:
+    internal: true
+    requires:
+      vars:
+        - AURORA_CLUSTER
+    cmd: ../aws/rds/aurora_endpoint.sh > .task/aurora-endpoint-{{.AURORA_CLUSTER}}
+    generates:
+      - .task/aurora-endpoint-{{.AURORA_CLUSTER}}
+    status:
+      - ! test -f .task/aurora-endpoint-{{.AURORA_CLUSTER}}
+
+  deploy-infinispan-crossdc:
+    desc: "Deploys Infinispan in both ROSA clusters"
+    internal: true
+    cmds:
+      - echo "Deploying Infinispan cluster with cross-site in ROSA clusters {{.ROSA_CLUSTER_NAME_1}} and {{.ROSA_CLUSTER_NAME_2}} (namespace={{.KC_ISPN_NAMESPACE}})"
+      - task: ispn:crossdc
+        vars:
+          OC_NAMESPACE_1: "{{.KC_ISPN_NAMESPACE}}"
+          OC_NAMESPACE_2: "{{.KC_ISPN_NAMESPACE}}"
+          CROSS_DC_HOT_ROD_PASSWORD: "{{.RS_HOT_ROD_PASSWORD}}"
+
+
+  fetch-ocp-hostname-prefix:
+    internal: true
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME
+    cmd: KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}" oc get route/console -n openshift-console -o jsonpath={.spec.host} | cut -d . -f 2- > ./.task/ocp-prefix-{{.ROSA_CLUSTER_NAME}}
+    generates:
+      - .task/ocp-prefix-{{.ROSA_CLUSTER_NAME}}
+    preconditions:
+      - test -f {{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}
+    status:
+      - ! test -f .task/ocp-prefix-{{.ROSA_CLUSTER_NAME}}
+
+  patch-keycloak-image:
+    internal: true
+    requires:
+      vars:
+        - ROSA_CLUSTER_NAME
+    cmds:
+      - envsubst < ../minikube/keycloak/operator-patch.yaml > .task/operator-patchfile-{{.ROSA_CLUSTER_NAME}}.yaml
+      - KUBECONFIG="{{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}" oc patch deployment keycloak-operator -n {{.KC_NAMESPACE_PREFIX}}keycloak --patch-file .task/operator-patchfile-{{.ROSA_CLUSTER_NAME}}.yaml
+    status:
+      - test "{{.KC_CONTAINER_IMAGE}}" == ""
+    preconditions:
+      - test -f {{.ISPN_DIR}}/.task/kubecfg/{{.ROSA_CLUSTER_NAME}}
+
+  install-keycloak:
```

Comment:
Can we move this to `provision/minikube/Utils.yaml` and reuse it in both `minikube/Taskfile` and `openshift/Taskfile`? It does not need to be done in this PR, just curious if you thought about this and whether there are some limitations. I can try to do this tomorrow.

Reply:
I tried to move the `keycloak` task from `openshift/Taskfile.yaml` to another taskfile to be included by both, but `isup.sh` was giving me an issue because it was reading the `.env` file from `openshift/.env` and failing. I'm not sure if it is worth the effort since it requires some changes; at the end of the day, it depends on how often the helm chart changes.